### ISSUE TITLE
Accessibility unit tests frequently fail, crash, or time out

### ISSUE THREAD

**Issue by Loirooriol** [servo/servo#47911]
I keep seeing intermittent problems with accessibility unit tests with `./mach test-unit -r`.

Examples:

```
        SLOW [>  5.000s] servo::accessibility test_activate_accessibility_after_layout
     SIGSEGV [   6.436s] servo::accessibility test_activate_accessibility_after_layout
  stdout ───

    running 1 test
    ┌ Accessibility Tree
    │  ├─ #0: GenericContainer (html_tag: "html")
    │  │         bounds: Rect { x0: 0.0, y0: 0.0, x1: 500.0, y1: 8.0 }
    │  │         children: [#1, #2]
    │  │  ├─ #1: GenericContainer (html_tag: "head")
    │  │  └─ #2: RootWebArea (html_tag: "body")
    │  │  │         bounds: Rect { x0: 8.0, y0: 8.0, x1: 492.0, y1: 8.0 }
    test test_activate_accessibility_after_layout ... ok

    test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 24 filtered out; finished in 0.74s
    
  stderr ───
    Redirecting call to abort() to mozalloc_abort
    

    (test aborted with signal 11: SIGSEGV)

────────────
     Summary [   6.494s] 994 tests run: 993 passed, 1 failed, 0 skipped
     SIGSEGV [   6.436s] servo::accessibility test_activate_accessibility_after_layout
error: test run failed
```

```
        SLOW [>  5.000s] servo::accessibility test_accessibility_bounds_updated_after_renderer_scroll
        SLOW [> 10.000s] servo::accessibility test_accessibility_bounds_updated_after_renderer_scroll
        SLOW [> 15.000s] servo::accessibility test_accessibility_bounds_updated_after_renderer_scroll
 TERMINATING [> 20.000s] servo::accessibility test_accessibility_bounds_updated_after_renderer_scroll
     TIMEOUT [  20.035s] servo::accessibility test_accessibility_bounds_updated_after_renderer_scroll
  stdout ───

    running 1 test
    ┌ Accessibility Tree
    │  ├─ #0: GenericContainer (html_tag: "html")
    │  │         bounds: Rect { x0: 0.0, y0: 0.0, x1: 500.0, y1: 2016.0 }
    │  │         children: [#1, #2]
    │  │  ├─ #1: GenericContainer (html_tag: "head")
    │  │  ├─ #2: RootWebArea (html_tag: "body")
    │  │  │         bounds: Rect { x0: 8.0, y0: 8.0, x1: 492.0, y1: 2008.0 }
    │  │  │         children: [#3, #4]
    │  │  │  ├─ #3: Main (html_tag: "main")
    │  │  │  │         bounds: Rect { x0: 10.0, y0: 100.0, x1: 110.0, y1: 150.0 }
    │  │  │  │         children: [#5]
    │  │  │  │  └─ #5: TextRun
    │  │  │  └─ #4: GenericContainer (html_tag: "div")
    │  │  │  │         bounds: Rect { x0: 8.0, y0: 8.0, x1: 2008.0, y1: 2008.0 }

    (test timed out)

────────────
     Summary [  20.122s] 994 tests run: 993 passed, 1 timed out, 0 skipped
     TIMEOUT [  20.035s] servo::accessibility test_accessibility_bounds_updated_after_renderer_scroll
error: test run failed
```
